SmartDeviceLink/public/SDLFileManager.h

```diff
@@ -136,6 +136,19 @@ typedef void (^SDLFileManagerStartupCompletionHandler)(BOOL success, NSError *__
  */
 - (void)uploadFiles:(NSArray<SDLFile *> *)files completionHandler:(nullable SDLFileManagerMultiUploadCompletionHandler)completionHandler NS_SWIFT_NAME(upload(files:completionHandler:));
 
+/**
+ * Check if an SdlFile needs to be uploaded to Core or not.
+ * It is different from hasUploadedFile() because it takes isStaticIcon and overwrite properties into consideration.
+ * ie, if the file is static icon, the method always returns false.
+ * If the file is dynamic, it returns true in one of these situations:
+ * 1) the file has the overwrite property set to true
+ * 2) the file hasn't been uploaded to Core before.
+ *
+ * @param file the SdlFile that needs to be checked
+ * @return boolean that tells whether file needs to be uploaded to Core or not
+ */
```

Comment:
```suggestion
/**
 * Check if an SDLFile needs to be uploaded to Core or not. This method differs from hasUploadedFile() because it takes the `isStaticIcon` and `overwrite` properties into consideration.
 *
 * For example, if the file is static icon, the method always returns false.
 *
 * If the file is dynamic, it returns true in one of these situations: 1) the file has the overwrite property set to true, 2) the file hasn't been uploaded to Core before.
 *
 * @param file the SDLFile that needs to be checked
 * @return BOOL that tells whether file needs to be uploaded to Core or not
 */
```